File: java/patterns-use-cases/event-processing-transactional-handlers/src/main/java/my/example/utils/Stubs.java
package my.example.utils;

import my.example.types.SocialMediaPost;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.UUID;

public class Stubs {
    private static final Logger logger = LogManager.getLogger(Stubs.class);

    public static String createPost(String userId, SocialMediaPost post) {
        String postId = UUID.randomUUID().toString();
        logger.info("Creating post {} for user {}", postId, userId);
        return postId;
    }

    public static String getPostStatus(String postId) {
        if (Math.random() < 0.8) {
            logger.info("Content moderation for post {} is still pending... Will check again in 5 seconds", postId);
            return "PENDING";
        } else {
            logger.info("Content moderation for post {} is done", postId);
            return "DONE";
        }
    }

    public static void updateUserFeed(String userId, String postId) {
        logger.info("Updating user feed for user {} with post {}", userId, postId);
    }
}

File: typescript/patterns-use-cases/src/eventtransactions/utils/stubs.ts
import { randomUUID } from "node:crypto";

export type SocialMediaPost = {
  content: string;
  metadata: string;
};

export const PENDING = "PENDING";
export const DONE = "DONE";

export async function createPost(
  userId: string,
  post: SocialMediaPost
): Promise<string> {
  const postId = randomUUID().toString();
  console.info(
    `Created post ${postId} for user ${userId} with content: ${post.content}`
  );
  return postId;
}

export async function getPostStatus(postId: string): Promise<string> {
  if (Math.random() < 0.8) {
    console.info(
      `Content moderation for post ${postId} is still pending... Will check again in 5 seconds`
    );
    return PENDING;
  } else {
    console.info(`Content moderation for post ${postId} is done`);
    return DONE;
  }
}

export async function updateUserFeed(user: string, postId: string) {
  console.info(`Updating the user feed for user ${user} with post ${postId}`);
}
